plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaSecretManager.java

```diff
@@ -42,23 +46,31 @@ public KafkaSecretManager(String agentName, KafkaTransportBuilder settings) {
         this.consumerTopic = settings.getConsumerTopic();
         this.consumerKey = settings.getConsumerKey();
         this.consumerPartition = settings.getConsumerPartition();
+        this.timeout = timeout;
     }
 
     public static String getConnectionSecret(String agentName) {
         return AGENT_SECRET.mac(agentName);
     }
 
-    public void waitForValidAgent() throws RemotingKafkaConfigurationException {
+    public boolean waitForValidAgent() throws RemotingKafkaConfigurationException {
         initHandshake();
-        waitForSecret();
+        return waitForSecret();
     }
 
-    private void waitForSecret() {
+    private boolean waitForSecret() {
         String connectionSecret = getConnectionSecret(agentName);
         String agentSecret = "";
         TopicPartition partition = new TopicPartition(consumerTopic, consumerPartition);
         consumer.assign(Arrays.asList(partition));
+        long start = System.currentTimeMillis();
         while (true) {
+            long current = System.currentTimeMillis();
+            if (current - start > timeout) {
+                LOGGER.warning("Timeout for waiting a secret from agent, restarting the agent...");
```

Comment:
Just "startup failed". You cannot say for sure whether and when Jenkins decides to restart it (e.g. it may be shutting down by this time)